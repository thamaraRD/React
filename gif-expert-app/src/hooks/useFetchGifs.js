import { useEffect, useState } from "react";
import { getGifApi } from "../helpers/getGifApi";

export const useFetchGifs = ( category ) => {

  const [images, setImages] = useState([]);
  const [loader, setLoader] = useState(true);
    
  const getImages = async () => {
      const getImages = await getGifApi(category);
      setImages(getImages);
      setTimeout(() => {
        setLoader(false);
      }, 0.5);
  };

  useEffect(() => {
    
    getImages();
  // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);
  
  return {
    images,
    loader
  }


};
