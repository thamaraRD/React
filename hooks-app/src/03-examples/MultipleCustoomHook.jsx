import React from 'react'
import { useFetch } from '../hooks/useFetch';

export const MultipleCustoomHook = () => {

  const { data, isLoading, hasError } = useFetch('https://api.breakingbadquotes.xyz/v1/quotes');
  console.log( { data, isLoading, hasError } );
  

return (
    <div>
        <h1>BreakingBad Quotes</h1>
        <hr />
    </div> 
    )};
