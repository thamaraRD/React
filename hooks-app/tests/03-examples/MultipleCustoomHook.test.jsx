import { fireEvent, render, screen } from "@testing-library/react";
import { MultipleCustoomHook } from "../../src/03-examples/MultipleCustoomHook";
import { useCounter } from "../../src/hooks/useCounter";
import { useFetch } from "../../src/hooks/useFetch";

jest.mock('../../src/hooks/useFetch');
jest.mock('../../src/hooks/useCounter');


describe('haciendo testing en el componente <MultipleCustoomHook />', () => { 
  
  const mockIncrement = jest.fn();

  useCounter.mockReturnValue({
   counter: 1,
   increment: mockIncrement
  });

  //Asegurar que cada una de las pruebas a funciones queden
  //reseteadas a su stado inicial
  beforeEach(() => {
    jest.clearAllMocks();
  })

  test('should de mostrar el componente por defecto', () => { 

    useFetch.mockReturnValue({
      data: null,
      isLoading: true,
      hasError: null
    })

    render( <MultipleCustoomHook /> );

    expect( screen.getByText('Loading...') );
    expect( screen.getByText('BreakingBad Quotes'))
    const nextQouteButton = (screen.getByRole('button', { name: 'Next Qoute' }))
    expect(nextQouteButton.disabled).toBeTruthy();
    // screen.debug();
  })

  test('should match to Snapshot', () => {

    const { container } = render ( <MultipleCustoomHook /> );
    expect(container).toMatchSnapshot();
  })

  test('should de mostrar un Quote', () => { 

    useFetch.mockReturnValue({
      data: [{ author: 'Thamara', quote: 'Go go go' }],
      isLoading: false,
      hasError: null
    })
    render( <MultipleCustoomHook /> )

    expect( screen.getByText('Go go go') ).toBeTruthy();
    expect( screen.getByText('- Thamara') ).toBeTruthy();
    const nextQouteButton = (screen.getByRole('button', { name: 'Next Qoute' }))
    expect(nextQouteButton.disabled).toBeFalsy();
    // screen.debug();
   })

  test('should de llamar la funciÃ³n incrementar', () => { 


    useFetch.mockReturnValue({
      data: [{ author: 'Thamara', quote: 'Go go go' }],
      isLoading: false,
      hasError: null
    })
    render( <MultipleCustoomHook /> )
    const nextQouteButton = (screen.getByRole('button', { name: 'Next Qoute' }));
    fireEvent.click( nextQouteButton );

    expect(mockIncrement).toHaveBeenCalled();

  })

});