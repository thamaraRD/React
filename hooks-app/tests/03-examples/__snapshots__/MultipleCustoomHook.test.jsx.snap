// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`haciendo testing en el componente <MultipleCustoomHook /> should match to Snapshot 1`] = `
<div>
  <h1>
    BreakingBad Quotes
  </h1>
  <hr />
  <div
    class="alert alert-warning text-center"
  >
    Loading...
  </div>
  <button
    class="btn btn-primary"
    disabled=""
  >
    Next Qoute
  </button>
</div>
`;
